### PR TITLE
cloudevent fields type checking adjustments

### LINKED ISSUE #95: is_binary should check more than just ce-specversion
## Expected Behavior
```python
from cloudevents.sdk.converters.binary import NewBinaryHTTPCloudEventConverter
cnvtr = NewBinaryHTTPCloudEventConverter()
assert not cnvtr.can_read({"ce-specversion": "1.0"})
```

## Actual Behavior
```python
cnvtr.can_read({"ce-specversion": "1.0"})
```
will return true because it only checks for ce-specversion and not other required versions.

### PR DESCRIPTION
**Pull request by cumason123**
Fixes #95, #94, #92 

## Changes
#92:
moved 
```python
from cloudevents.sdk.converters import is_binary, is_structured
```

into 
```python
from cloudevents.http import is_binary, is_structured
```

#94 
added
```python
from cloudevents.exceptions import CloudEventMissingRequiredFields, CloudEventTypeErrorRequiredFields
```

#95:
[BinaryHTTPCloudEventConverter.can_read](https://github.com/cloudevents/sdk-python/blob/is_binary_can_read_fix/cloudevents/sdk/converters/binary.py)
now checks 4 required fields instead of just 1

## One line description for the changelog
cloudevent fields type checking adjustments

- [*] Tests pass
- [*] Appropriate changes to README are included in PR

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,20 @@ All notable changes to this project will be documented in this file.
...
 ## [1.0.0]
 ### Added
+- Update types and handle data_base64 structured ([#34])
 - Added a user friendly CloudEvent class with data validation ([#36])
 - CloudEvent structured cloudevent support ([#47])
+- Separated http methods into cloudevents.http module ([#60])
+- Implemented to_json and from_json in http module ([#72])
+
+### Fixed
+- Fixed top level extensions bug ([#71])
```
Comment (grant) [inline CHANGELOG.md:20]: I think we usually keep changelog PRs separate from other PRs.

File: cloudevents/http/__init__.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 import typing
 
 from cloudevents.http.event import CloudEvent
+from cloudevents.http.event_type import is_binary, is_structured
```
Comment (grant) [inline cloudevents/http/__init__.py:18]: Why do we need this import?

File: cloudevents/sdk/converters/util.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import typing
+
+
+def has_binary_headers(headers: typing.Dict[str, str]) -> bool:
+    return (
+        "ce-specversion" in headers
+        and "ce-source" in headers
+        and "ce-type" in headers
+        and "ce-id" in headers
+    )
```
Comment (grant) [inline cloudevents/sdk/converters/util.py:10]: Do we really need a separate file? Can we put this method in an existing file?

File: cloudevents/sdk/event/base.py [hunk 4]
```diff
@@ -193,7 +198,10 @@ def Set(self, key: str, value: object):
 
     def MarshalJSON(self, data_marshaller: types.MarshallerType) -> str:
         if data_marshaller is None:
-            data_marshaller = lambda x: x  # noqa: E731
+
+            def data_marshaller(x):
+                return x  # noqa: E731
+
```
Comment (grant) [inline cloudevents/sdk/event/base.py:204]: Not sure why this change is made.

File: cloudevents/tests/test_http_events.py [hunk 5]
```diff
@@ -73,9 +75,13 @@ def post(url, headers, data):
 
 @app.route("/event", ["POST"])
 async def echo(request):
-    decoder = None
     if "binary-payload" in request.headers:
-        decoder = lambda x: x
+
+        def decoder(x):
+            return x
```
Comment (grant) [inline cloudevents/tests/test_http_events.py:81]: ditto, not sure why we are changing from a lambda here.

File: cloudevents/http/__init__.py (hunk 2 above)
Comment (cumason123) [reply cloudevents/http/__init__.py:18]: exporting is_binary and is_structured under a more convenient name. With that import I can now do
```python
from cloudevents.http import is_binary, is_structured
```

without I'd need to do
```python
from cloudevents.http.event_type import is_binary, is_structured
```

File: cloudevents/sdk/converters/util.py (hunk 3 above)
Comment (cumason123) [reply cloudevents/sdk/converters/util.py:10]: I wanted both [BinaryHTTPCloudEventConverter](https://github.com/cloudevents/sdk-python/blob/master/cloudevents/sdk/converters/binary.py) and [JSONHTTPCloudEventConverter](https://github.com/cloudevents/sdk-python/blob/master/cloudevents/sdk/converters/structured.py) to have access to this method. I could move both files into one file and move that function into that file. That would just be up to preferences I think.

File: cloudevents/sdk/event/base.py (hunk 4 above)
Comment (cumason123) [reply cloudevents/sdk/event/base.py:204]: honestly, my auto formatter kept doing that... I'll look into either fixing my formatter and reverting the change

File: CHANGELOG.md (hunk 1 above)
Comment (cumason123) [reply CHANGELOG.md:20]: I'll revert this in that case

File: cloudevents/http/__init__.py (hunk 2 above)
Comment (grant) [reply cloudevents/http/__init__.py:18]: Ah thanks.

File: cloudevents/sdk/converters/util.py (hunk 3 above)
Comment (grant) [reply cloudevents/sdk/converters/util.py:10]: Sounds good.
